thunder.install

```diff
@@ -426,3 +427,115 @@ function thunder_update_8106() {
 
   return $updateLogger->outputHtml();
 }
+
+/**
+ * Installs the thunder_riddle module.
+ */
+function thunder_update_8107() {
+
+  $thunderUpdater = \Drupal::service('thunder_updater');
+
+  /** @var \Drupal\thunder\ThunderUpdateLogger $updateLogger */
+  $updateLogger = \Drupal::service('logger.thunder_update');
+  $updateLogger->cleanLogs();
+
+  $successful = TRUE;
+  if (\Drupal::service('module_installer')->install(['thunder_riddle'])) {
+
+    // Import Riddles.
+    \Drupal::service('media_riddle_marketplace')->createMediaEntities();
+
+    /** @var \Drupal\paragraphs\Entity\Paragraph[] $paragraphs */
+    $paragraphs = \Drupal::entityTypeManager()->getStorage('paragraph')
+      ->loadByProperties([
+        'type' => 'paragraphs_riddle_marketplace',
+      ]);
+
+    $paragraphFields = [];
+    foreach ($paragraphs as $oldParagraph) {
+
+      $fieldName = $oldParagraph->get('parent_field_name')->get(0)->getString();
+
+      $host = $oldParagraph->getParentEntity();
+      if ($host) {
+
+        $paragraphFields[] = $host->getEntityTypeId() . '.' . $host->bundle() . '.' . $fieldName;
+
+        foreach ($host->get($fieldName)->getValue() as $delta => $field) {
+          if ($field['target_id'] == $oldParagraph->id()) {
+
+            /* @var \Drupal\media_entity\Entity\Media */
+            $media = \Drupal::entityTypeManager()->getStorage('media')
+              ->loadByProperties([
+                'bundle' => 'riddle',
+                'field_riddle_id' => end(explode('/', $oldParagraph->get('field_link')->uri)),
+              ]);
+
+            $media = current($media);
+
+            $paragraph = Paragraph::create([
+              'type' => 'riddle',
+              'field_riddle' => $media->id(),
+            ]);
+            $paragraph->save();
+
+            $host->get($fieldName)->set($delta, [
+              'target_id' => $paragraph->id(),
+              'target_revision_id' => $paragraph->getRevisionId(),
+            ]);
+
+            if ($host->save()) {
+              $updateLogger->warning(t('Converted @riddle in @name.', ['@riddle' => $media->label(), '@name' => $host->label()]));
+              $oldParagraph->delete();
+            }
+            else {
+              $successful = FALSE;
```

Comment:
This should be logged too. Maybe it would be best warning here and info on successful removal. Also if old paragraph is not deleted, that should be logged too.